OpenRA.Mods.TS/Traits/Buildings/LowBridge.cs

```diff
@@ -0,0 +1,50 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2016 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.TS.Traits
+{
+	[Desc("Bridge actor that can't be passed underneath.")]
+	class LowBridgeInfo : ITraitInfo, Requires<BuildingInfo>
+	{
+		public readonly string TerrainType = "Road";
+
+		public object Create(ActorInitializer init) { return new LowBridge(init.Self, this); }
+	}
+
+	class LowBridge : INotifyCreated
+	{
+		readonly LowBridgeInfo info;
+
+		public LowBridge(Actor self, LowBridgeInfo info)
+		{
+			this.info = info;
+		}
+
+		void INotifyCreated.Created(Actor self)
```

Comment:
This will need to be done on AddedToWorld, and then a matching RemovedFromWorld should remove the custom terrain.

I'm not sure why I missed this earlier, sorry.